align/affineGap.go

```diff
@@ -7,27 +7,29 @@ import (
 	"log"
 )
 
-// the data structure is a 3d slice where the first index is 0,1,2 and represents
-// the match, gap in x (first seq), and gap in y (second seq).
-func initAffineScoringAndTrace(firstSeqLen int, secondSeqLen int) ([][][]int64, [][][]ColType) {
-	m := make([][][]int64, 3)
+// the trace data structure is a 3d slice where the first index is 0,1,2 and represents the match, gap in x (first seq), and gap in y (second seq).
+// m used to have the same data structure as trace, but has been simplified into a 2d slice, where the second index for mColumn is removed in order to recycle memory by rows
+func initAffineScoringAndTrace(firstSeqLen int, secondSeqLen int) ([][]int64, [][]int64, int, [][][]ColType) {
```

Comment:
I'm not used to seeing/interpreting stacked slices like this, so just to be sure, is [ ][ ][ ] where we have a slice of slices of slices??? Like [ [ [ABC][ABC] ]  [ [DEF][DEF] ]   ]  ?

Reply:
Yes! I think of it as a 3D matrix, or a gridded diagram with 3 axes (x,y,z) and there is a number in each box. But in this case I just think of a 2D matrix*3 since the outermost [] has only 3 possible values (0,1,2)